### PR TITLE
Use chapter data in navigation

### PR DESCRIPTION
**Pull request by DanielKuhn**
REDMINE-17300

### CODE REVIEW

File: entry_types/scrolled/lib/pageflow_scrolled/seeds.rb [hunk 1]
```diff
@@ -53,21 +50,34 @@ def say_creating_scrolled_entry(entry)
...
       end
     end
 
-    def create_content_element(section, content_element_config, position)
+    def create_content_element(section, content_element_config)
       section.content_elements.create!(
         type_name: content_element_config['type'],
-        configuration: content_element_config['props'].except('anchor'),
-        position: position
+        configuration: content_element_config['props']
```
Comment (tf) [inline entry_types/scrolled/lib/pageflow_scrolled/seeds.rb]: Why is position no longer required?

File: entry_types/scrolled/package/src/frontend/Chapter.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ import Section from './Section';
 
 export default function Chapter(props) {
   return (
-    <div id={'chapter' + props.permaId}>
+    <div id={'chapter-'+props.permaId}>
```
Comment (tf) [inline entry_types/scrolled/package/src/frontend/Chapter.js]: Just as a aside: A template string, i.e. `` `chapter-${props.permaId}` ``, would be more modern/idiomatic, I think

File: entry_types/scrolled/package/src/frontend/navigation/AppHeader.js [hunk 3]
```diff
@@ -51,11 +58,27 @@ export default function AppHeader() {
...
+    setActiveChapterLink(chapterLinkId);
     setMobileNavHidden(true);
   };
 
+  function renderMenuLinks(chapters) {
+    return chapters.map((chapter, index) => {
+      const chapterIndex = index + 1;
+      const chapterLinkId = 'chapterLink' + chapterIndex;
+      return (
+        <li key={index} className={styles.chapterListItem}>
```
Comment (tf) [inline entry_types/scrolled/package/src/frontend/navigation/AppHeader.js]: Also just as aside: Whenever possible it's best to use keys that represent stable identity. React uses keys to determine whether an element has been inserted. If you used the chapter id as key, for example, and then a chapter gets inserted after the first one, React can actually reuse the DOM nodes of all chapters and just inserts one new DOM node. Using an index as a key causes React to to view the inserted chapter as the new way the second chapter now looks. This causes all following chapter nodes to be updated, basically updating the nth chapter with the contents of the (n-1)th.

Comment (DanielKuhn) [reply entry_types/scrolled/package/src/frontend/navigation/AppHeader.js]: 👍

**Pull request merged**